CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+cmake_minimum_required(VERSION 3.25)
```

Comment:
IMO, the CMake changes should be kept in only #1947, it's out of scope for this PR.

Reply:
Yep; fully agree.  Thanks for the feedback!